pre_commit_hooks/insert_license.py

```diff
@@ -194,9 +200,15 @@ def license_not_found(remove_header, license_info, src_file_content, src_filepat
         index = 0
         for line in src_file_content:
             stripped_line = line.strip()
-            # Special treatment for shebang, encoding and empty lines when at the beginning of the file
+            # Special treatment for user provided regex,
+            # or shebang, file encoding directive,
+            # and empty lines when at the beginning of the file.
             # (adds license only after those)
-            if stripped_line.startswith("#!") \
+            if after_regex is not None and after_regex != "":
```

Comment:
A simple `if after_regex:` would cover both cases of `None` and the empty string